### PR TITLE
[aot] C_API behavior test

### PR DESCRIPTION
**Pull request by damnkk**
Issue: #

### Brief Summary
Added run out of memory test case.

### CODE REVIEW

File: taichi/rhi/vulkan/vulkan_api.cpp [hunk 1]
```diff
@@ -515,7 +515,9 @@ IVkBuffer create_buffer(VkDevice device,
 
   VkResult res = vmaCreateBuffer(allocator, buffer_info, alloc_info,
                                  &buffer->buffer, &buffer->allocation, nullptr);
-  assert(res != VK_ERROR_OUT_OF_DEVICE_MEMORY);
+  if (res == VK_ERROR_OUT_OF_DEVICE_MEMORY) {
```
Comment (PENGUINLIONG) [inline taichi/rhi/vulkan/vulkan_api.cpp:518]: // FIXME: (damnkk) Should be removed when RHI error codes are ready

**Review by PENGUINLIONG — CHANGES_REQUESTED** [review]

**Comment by PENGUINLIONG** [comment]
There is an on-going rafactorization to add error codes to all RHI interfaces. We don't have it for now so I'm not sure if it's a proper way to report OOM errors for the release. Or we just wait for a future commit of OOM RHI error to be merged? wdyt @bobcao3 @ailzhang

File: c_api/tests/c_api_behavior_test.cpp [hunk 2]
```diff
@@ -117,39 +117,51 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
...
-        TiMemoryAllocateInfo allocate_info;
-        allocate_info.size = 1024;
-        allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT << i;
-        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
-        TI_ASSERT(memory != TI_NULL_HANDLE);
-        ti_free_memory(runtime, memory);
-      }
+    if (!ti::is_arch_available(arch)) {
+      TI_WARN("arch {} is not supported, so this test is skipped", arch);
+    }
```
Comment (ailzhang) [inline c_api/tests/c_api_behavior_test.cpp:122]: nit: early return after warning?

File: c_api/tests/c_api_behavior_test.cpp [hunk 3]
```diff
@@ -117,39 +117,51 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
...
 
-      // runtime and allocate_info are both null
-      {
-        ti_allocate_memory(TI_NULL_HANDLE, nullptr);
-        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
-      }
+    // Attempt to allocate memory with size of 1024
+    TiRuntime runtime = ti_create_runtime(arch);
```
Comment (ailzhang) [inline c_api/tests/c_api_behavior_test.cpp:125]: nit: better split these to 5 smaller / separate tests to keep tests minimal and isolated ;)

**Comment by ailzhang** [comment]
IMHO it doesn't have to be blocked by the missing OOM error code in rhi refactorization as long as we add a FIXME and fix it right after the merge. Adding these tests will also help make sure the refactorization goes well.

**Review by PENGUINLIONG — APPROVED** [review]
lgtm then. thank you!

**Pull request merged**